go/vt/wrangler/vdiff.go

```diff
@@ -1051,29 +1045,16 @@ func (td *tableDiffer) compare(sourceRow, targetRow []sqltypes.Value, cols []com
 			continue
 		}
 		compareIndex := col.colIndex
-		// This detects if we are using weight_string() to compare this (text) column.
-		// If either source or target weight_string is null we fallback to a byte compare for text columns
-		if !sourceRow[col.weightStringIndex].IsNull() && sourceRow[col.colIndex].IsText() &&
-			!targetRow[col.weightStringIndex].IsNull() && targetRow[col.colIndex].IsText() &&
-			col.weightStringIndex > col.colIndex {
-			compareIndex = col.weightStringIndex
-		}
 		var c int
 		var err error
-		if sourceRow[compareIndex].IsText() && targetRow[compareIndex].IsText() {
-			srowBytes, err := sourceRow[compareIndex].ToBytes()
-			if err != nil {
-				return 0, err
-			}
-			trowBytes, err := targetRow[compareIndex].ToBytes()
-			if err != nil {
-				return 0, err
-			}
-			c = bytes.Compare(srowBytes, trowBytes)
+		var collationID collations.ID
+		// if the collation is nil or unknown, use binary collation to compare as bytes
+		if col.collation == nil {
+			collationID = collations.CollationBinaryID
 		} else {
-			// TODO(king-11) make collation aware
-			c, err = evalengine.NullsafeCompare(sourceRow[compareIndex], targetRow[compareIndex], collations.Unknown)
+			collationID = col.collation.ID()
 		}
+		c, err = evalengine.NullsafeCompare(sourceRow[compareIndex], targetRow[compareIndex], collationID)
```

Comment:
Nice to be able to just use `NullsafeCompare` here and get rid of the `weight_string()` related code!